### PR TITLE
Add new handler: `src/handlers/lineargradient.js`

### PR DESCRIPTION
**Pull request by Raruto**
## Preview

Add support for graduated chart profiles colors:

image

# Sample usage:

```js

L.control.Elevation({
  altitude: true,
  distance: true,
  handlers: [ 'Altitude', 'Distance', 'LinearGradient', ],
  linearGradient: {
    attr: 'z',
    path: 'altitude',
    range: { 0.0: '#008800', 0.5: '#ffff00', 1.0: '#ff0000' },
    min: 'elevation_min',
    max: 'elevation_max',
  },
})
```

## Further information:

- Make use of [`<linearGradient>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/linearGradient) 👉 when **`preferCanvas: false`**  
- Make use of [`CanvasGradient`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient) 👉 when **`preferCanvas: true`**

Closes: https://github.com/Raruto/leaflet-elevation/issues/251 
Partially handles: https://github.com/Raruto/leaflet-elevation/issues/232

### CODE REVIEW

File: src/components/d3.js [hunk 1]
```diff
@@ -609,8 +612,13 @@ export const Chart = ({
...
-		if(/Mac|iPod|iPhone|iPad/.test(navigator.platform) && /AppleWebkit/i.test(navigator.userAgent)) {
+		// Fix: https://github.com/Raruto/leaflet-elevation/issues/123
+		// Fix: https://github.com/Raruto/leaflet-elevation/issues/232
+		if (
+			/Mac|iPod|iPhone|iPad/.test(navigator.platform) &&
+			/AppleWebkit/i.test(navigator.userAgent) &&
+			!/Chrome/.test(navigator.userAgent)
+		) {
 			canvas   .style("transform", "translate(" + margins.left + "px," + margins.top + "px)");
 		}
```
Comment (Raruto) [inline src/components/d3.js:623]: @hupe13 do you confirm that this is the patch you proposed in here? 👉 https://github.com/Raruto/leaflet-elevation/issues/232

**Comment by Raruto** [comment]
@jtpio it's definitely not perfect, but please let me know if you already see any room for improvement.

**Comment by hupe13** [comment]
> do you confirm that this is the patch you proposed in here? 👉 https://github.com/Raruto/leaflet-elevation/issues/232

Yes it works right now on Chrome and Safari. 

But this problem still exists on Safari:
grafik
This is your /examples/leaflet-elevation.html. I set `preferCanvas: true,`.

**Comment by jtpio** [comment]
> @jtpio it's definitely not perfect (performances ?), but please let me know if you already see any room for improvement.

Thanks @Raruto this is looking pretty good already!

Just a minor note on the screenshot posted above. It looks like the red color should probably be associated with the steeper sections of the elevation graph?

image

File: src/handlers/lineargradient.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      gradient.addColorStop = function(offset, color) { gradient.append('svg:stop').attr('offset', offset).attr('stop-color', color) };
+      path.attr('fill', 'url(#' + gradient_id + ')').classed('area', false);
+    }
+
+    // Generate gradient for each segment picking colors from palette
+    for (let i = 0, data = this._data; i < data.length; i++) {
+      gradient.addColorStop((i) / data.length, palette.getRGBColor(data[i][attr]));
+    }
+
+  });
```
Comment (Raruto) [inline src/handlers/lineargradient.js:105]: @jtpio ref: https://github.com/Raruto/leaflet-elevation/pull/257#issuecomment-1556613584

I tried to write an algorithm as similar as possible to the one used to generate the `hotline` colors:

https://github.com/Raruto/leaflet-elevation/blob/e0c68cba9a71d140e4c5a4179c51ae34588b7327/libs/leaflet-hotline.js#L238-L266

but as you can see, I also believe that there is a shift in here: `src/handlers/lineargradient.js`.

By chance, are you able to take a look at how to fix it?

It took me quite a bit of work to get to this point (and right now I don't think I'll be using this code any time soon..)

Comment (Raruto) [reply src/handlers/lineargradient.js:105]: meanwhile I'm going to merging this, feel free to open a new pull request to improve or fix this

**Comment by Raruto** [comment]
Feature released in **v2.5.0**

**Comment by jtpio** [comment]
Nice thanks @Raruto!

**Pull request merged**